src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -418,28 +410,36 @@ Either<ReplicateStatusUpdateError, TaskNotificationType> updateReplicateStatusWi
         log.info("Replicate update request [status:{}, chainTaskId:{}, walletAddress:{}, details:{}]",
                 statusUpdate.getStatus(), chainTaskId, walletAddress, statusUpdate.getDetailsWithoutLogs());
 
-        final ReplicateStatusUpdateError error = canUpdateReplicateStatus(chainTaskId, walletAddress, statusUpdate, updateReplicateStatusArgs);
+        final Optional<ReplicatesList> oReplicatesList = getReplicatesList(chainTaskId);
+        final Optional<Replicate> oReplicate = oReplicatesList
+                .flatMap(replicatesList -> replicatesList.getReplicateOfWorker(walletAddress));
+        if (oReplicatesList.isEmpty() || oReplicate.isEmpty()) {
+            log.error("Cannot update replicate, could not get replicate [chainTaskId:{}, UpdateRequest:{}]",
+                    chainTaskId, statusUpdate);
+            return Either.left(ReplicateStatusUpdateError.UNKNOWN_REPLICATE);
+        }
+        final ReplicatesList replicatesList = oReplicatesList.get();
+        final Replicate replicate = oReplicate.get();
+        ReplicateStatus newStatus = statusUpdate.getStatus();
```

Comment:
@mcornaton  isn't here but I'm sure he'd love to see a final here

Reply:
done, see 41199b6